### DISCUSSION TITLE
How can client side code get notified when a new page is added or page is edited?

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress/core#661] [Q&A] [answered]
I am making a blog plugin. And I would like to add a listener where a new page is created.

Since we already support `routeMeta`, my blog plugin are now:

1. writing author, category, time and such info in to routeMeta
2. Use `router.getRoutes()` and collect info of all pages and provide an article list during setup.
3. Use `inject` to get the aritlcle list in the components

The above is working pretty fine. But since there isn't a reactive `routes` in the router public api (only `getRoute()`), the setup function won't invoke and update aricle list when:

1. user modify frontmatter
2. user deletes page
3. user adds page

I am wondering if there is a solution here to make the above changes support HMR and get updates in devServer.

Any help would be appreciated.

**Comment by meteorlxy** [comment, marked as answer]
Try triggering the HMR manually? 🤔  As I remember, vue-router allows to adding routes dynamically. Not sure about deleting routes...

**Reply by Mister-Hope** [reply]
> vue-router allows to adding routes dynamically. Not sure about deleting routes...

Yes, but routes are not reactive,

I finally handle HMR myself, and build a blog plugin for vuepress2
